src/index.ts

```diff
@@ -80,53 +74,65 @@ function decrypt<R>(password: string, text: string): Promise<R> {
  * @param {CryptoKey} key - CryptoKey to decrypt with
  * @param {EncryptionResult} payload - payload returned from an encryption method
  */
-function decryptWithKey<R>(
+async function decryptWithKey<R>(
   key: CryptoKey,
   payload: EncryptionResult,
 ): Promise<R> {
   const encryptedData = Buffer.from(payload.data, 'base64');
   const vector = Buffer.from(payload.iv, 'base64');
-  return crypto.subtle
-    .decrypt({ name: 'AES-GCM', iv: vector }, key, encryptedData)
-    .then(function (result) {
-      const decryptedData = new Uint8Array(result);
-      const decryptedStr = Buffer.from(decryptedData).toString('utf-8');
-      const decryptedObj = JSON.parse(decryptedStr);
-      return decryptedObj;
-    })
-    .catch(function (_error) {
-      throw new Error('Incorrect password');
-    });
+  const result = await crypto.subtle.decrypt(
+    { name: 'AES-GCM', iv: vector },
+    key,
+    encryptedData,
+  );
+
+  const decryptedData = new Uint8Array(result);
+  const decryptedStr = Buffer.from(decryptedData).toString('utf-8');
+
+  let decryptedObj;
+  try {
+    decryptedObj = JSON.parse(decryptedStr);
```

Comment:
You might want to move the `decrypt` call, and everything in-between, into this `try` block as well. I would guess that it throws if it can't decrypt? Not sure. But that would match the old error handling behaviour

Reply:
It looks like the test failures are related to this. But.... curious that they fail only for v16 :thinking: It's as though

Reply:
Fixed!